examples/plotting/file/clustering.py

```diff
@@ -0,0 +1,64 @@
+""" Example inspired by an example from the scikit-learn project:
+http://scikit-learn.org/stable/auto_examples/cluster/plot_cluster_comparison.html
+"""
+
+import time
+
+import numpy as np
+
+from sklearn import cluster, datasets
+from sklearn.neighbors import kneighbors_graph
+from sklearn.preprocessing import StandardScaler
```

Comment:
probably worth a `try...except ImportError` with a specific message saying this particular example requires `sklearn` (use `from __future__ import print_function`)